impl io::{Read,Write} for PtyMaster

cuviper opened the pull request:
`PtyMaster` acts like an owned file descriptor, even closing on `Drop`.
Implementing `io::Read` and `io::Write` lets it be used directly in
standard I/O operations.

asomers commented:
This change makes sense to me.  However, please split that test method up.  It's too big, and tests too much.  I'm also going to request a review from @Susurrus because he's more knowledgeable about pseudoterminals than I am.

cuviper commented:
>  However, please split that test method up. It's too big, and tests too much.

I kept it together because the existing test code is all necessary setup to get a usable pair for my read/write tests. Even if I extracted that setup into some common function, the only difference from the current test would be the two asserts for `fd > 0`.

asomers commented:
> > However, please split that test method up. It's too big, and tests too much.
> 
> I kept it together because the existing test code is all necessary setup to get a usable pair for my read/write tests. Even if I extracted that setup into some common function, the only difference from the current test would be the two asserts for `fd > 0`.

My objection is that you're testing `io::Read` and `io::Write` in the same test, even though they're logically separate functionality.  Those should be in separate test methods, sharing a common setup function.

cuviper commented:
> Those should be in separate test methods, sharing a common setup function.

OK, I've now split that up.

cuviper commented:
I've rebased and squashed to one commit.

I think CI isn't triggering with the current GitHub outages, but I'll check back in a while and re-push if that still hasn't gone anywhere...

cuviper commented:
OK, CI is green now.

asomers commented:
bors retry

asomers commented:
bors retry
